web/concrete/src/Console/Command/ServiceCommand.php

```diff
@@ -0,0 +1,148 @@
+        $rc = 0;
+        try {
+            $manager = Core::make('Concrete\Core\Service\Manager\ServiceManager');
+            /* @var \Concrete\Core\Service\Manager\ServiceManager $manager */
+            $service = $manager->getService($input->getArgument('service'), $input->getOption('service-version'));
+            if ($service === null) {
+                $msg = 'Unknown web server handle: '.$input->getArgument('service');
+                $msg .= PHP_EOL;
+                $msg .= 'Valid handles: '.implode($manager->getExtensions());
+                throw new Exception($msg);
+            }
+            $operation = $input->getArgument('operation');
+            switch ($operation) {
+                case 'check':
+                    if ($this->checkConfiguration($service, $output) === false) {
+                        $rc = 2;
+                    }
+                    break;
+                case 'update':
+                    $this->updateConfiguration($service, $output);
+                    break;
+                default:
+                    throw new Exception('Invalid value of the operation argument (valid values: check or update');
+            }
+        } catch (Exception $x) {
+            $output->writeln('<error>'.$x->getMessage().'</error>');
+            $rc = 1;
+        }
+
+        return $rc;
+    }
+
+    public function checkConfiguration(ServiceInterface $service, OutputInterface $output)
+    {
+        $storage = $service->getStorage();
+        if (!$storage->canRead()) {
+            throw new Exception('Unable to read current server configuration for '.$service->getFullName());
+        }
+        $configuration = $storage->read();
+        $generator = $service->getGenerator();
+        $configurator = $service->getConfigurator();
+        $allOk = true;
+        foreach ($generator->getRules() as $ruleHandle => $rule) {
+            $shouldHave = $generator->ruleShouldBeEnabled($ruleHandle);
+            if ($shouldHave !== null) {
+                if ($shouldHave) {
+                    $output->write("Checking presence of rule $ruleHandle... ");
+                    if ($configurator->hasRule($configuration, $rule)) {
+                        $output->writeln('<info>found (ok).</info>');
+                    } else {
+                        $output->writeln('<error>NOT FOUND!</error>');
+                    }
+                } else {
+                    $output->write("Checking absence of rule $ruleHandle... ");
+                    if ($configurator->hasRule($configuration, $rule)) {
+                        $output->writeln('<error>FOUND!</error>');
+                    } else {
+                        $output->writeln('<info>not found (ok).</info>');
+                    }
+                }
+            }
+        }
+
+        return $allOk;
+    }
+
+    public function updateConfiguration(ServiceInterface $service, OutputInterface $output)
+    {
+        $storage = $service->getStorage();
```

Comment:
Some spacing in here and some comments would go really far for readability, my eyes cannot penetrate this block of code at all